model/UserManager.php

```diff
@@ -9,7 +9,13 @@ public function getUser(string $pseudo): object
         $query = $this->db->prepare('SELECT * FROM users WHERE pseudo = ?');
         $query->execute([$pseudo]);
 
-        return new User($query->fetch());
+        if (is_bool($query->fetch())) {
+            throw new \Exception("Ce pseudo n'existe pas.");
+        } else {
+            var_dump("test");
+
+            return new User($query->fetch());
+        }
```

Comment:
This can be refactored to not use fetch twice, remove the `else` (`throw` will interrupt the script flow, just like `return`, so you don't need the `else`, and replace `is_bool` with a more precise condition.

```suggestion
        if (false === ($data = $query->fetch())) {
            throw new \Exception("Ce pseudo n'existe pas.");
        }
        
        return new User($data);
```

Reply:
Thanks !